kotlinx-coroutines-core/common/src/channels/Broadcast.kt

```diff
@@ -107,8 +145,9 @@ private open class BroadcastCoroutine<E>(
     }
 
     override fun cancelInternal(cause: Throwable) {
-        _channel.cancel(cause.toCancellationException()) // cancel the channel
-        cancelCoroutine(cause) // cancel the job
+        val exception = cause.toCancellationException()
+        _channel.cancel(exception) // cancel the channel
+        cancelCoroutine(exception) // cancel the job
```

Comment:
Could you please elaborate why did you change `cancelCoroutine(cause)` to `cancelCoroutine(cause.toCancellationException())`?

Reply:
I've changed it for consistency with `ChannelCoroutine`, so that `cancelInternal` in both `BroadcastCoroutine` and `ChannelCoroutine` are the same (it is easier to compare them side-by-side this way). 

It should not really matter since the only method that can call `cancelInternal` with non-CancellationException cause is `JobSupport.cancel(cause: Throwable?)` which has been deprecated & hidden long time ago.